src/components/ResourceReview/EditResourceReview.tsx

```diff
@@ -86,81 +87,101 @@ export function EditResourceReview({
-                    <Text lineClamp={1} size="xs" color="dimmed">
-                      {modelVersionName}
-                    </Text>
-                  )}
-                </Stack>
-              </NextLink>
-              <Rating value={rating} onChange={handleRatingChange} />
-            </Group>
-            {createdAt && (
-              <Text size="xs">
-                Reviewed <DaysFromNow date={createdAt} />
-              </Text>
-            )}
-          </Stack>
+    <ResourceAccessWrap
+      entityType="ModelVersion"
+      entityId={modelVersionId}
+      fallback={
+        modelName ? (
+          <Alert>
+            <Text>
+              You cannot review the model{' '}
+              <Text weight="bold" component="span">
+                {modelName} {modelVersionName ? `(${modelVersionName})` : ''}
+              </Text>{' '}
+              as you have no access to it
+            </Text>
+          </Alert>
         ) : (
-          <Text>{name}</Text>
-        )}
-        {id && (
-          <>
-            <Card.Section>
-              <Divider />
-            </Card.Section>
-            <Stack>
-              {!editDetail ? (
-                <Text variant="link" onClick={toggleEditDetail} size="sm">
-                  <Group spacing={4} sx={{ cursor: 'pointer' }}>
-                    <IconChevronDown size={16} />{' '}
-                    <span>{!details ? 'Add' : 'Edit'} Review Comments</span>
-                  </Group>
-                </Text>
-              ) : (
-                <Form form={form} onSubmit={handleSubmit}>
-                  <Stack spacing="xs">
-                    <InputRTE
-                      name="details"
-                      includeControls={['formatting', 'link']}
-                      hideToolbar
-                      editorSize="sm"
-                      innerRef={commentRef}
-                      placeholder={`What did you think of ${modelName}?`}
-                      styles={{ content: { maxHeight: 500, overflowY: 'auto' } }}
-                      // withLinkValidation
-                    />
-                    <Group grow spacing="xs">
-                      <Button size="xs" variant="default" onClick={toggleEditDetail}>
-                        Cancel
-                      </Button>
-                      <Button
-                        size="xs"
-                        type="submit"
-                        loading={isLoading}
-                        variant={form.formState.isDirty ? undefined : 'outline'}
-                      >
-                        Save
-                      </Button>
-                    </Group>
+          <Alert>You cannot review the a resource you have no access to</Alert>
+        )
+      }
+    >
+      <Card p={8} withBorder>
+        <Stack spacing="xs">
+          {modelVersionId ? (
+            <Stack spacing={4}>
+              <Group align="center" position="apart">
+                <NextLink
```

Comment:
Please just use <Link> instead from `next/link`
```suggestion
                <Link
```